sklearn/utils/estimator_checks.py

```diff
@@ -454,8 +456,85 @@ def check_dict_unchanged(name, Estimator):
                               'Estimator changes __dict__ during %s' % method)
 
 
+def check_fit_adds_private_attributes_only(name, Estimator):
+    if name in ['SGDClassifier', 'Perceptron',
+                'PassiveAggressiveClassifier',
+                'LassoLarsIC', 'LassoLarsCV',
+                'LarsCV', 'GraphLassoCV',
+                'GradientBoostingRegressor',
+                'GradientBoostingClassifier',
+                'GaussianProcessRegressor',
+                'GaussianProcess', 'TSNE']:
+        # these estimators are not following new rule
+        return 0
+    rnd = np.random.RandomState(0)
+    X = 3 * rnd.uniform(size=(20, 3))
+    y = X[:, 0].astype(np.int)
+    y = multioutput_estimator_convert_y_2d(name, y)
+    estimator = Estimator()
+    set_testing_parameters(estimator)
+
+    if hasattr(estimator, "n_components"):
+        estimator.n_components = 1
+    if hasattr(estimator, "n_clusters"):
+        estimator.n_clusters = 1
+
+    set_random_state(estimator, 1)
+    dict_before_fit = estimator.__dict__.copy()
+    estimator.fit(X, y)
+
+    dict_after_fit = estimator.__dict__
+    # leave only attributes which have been set by fit
+    substracted_dicts_keys = [k for k in dict_after_fit.keys()
+                              if k not in dict_before_fit.keys()]
+
+    for val in substracted_dicts_keys:
+        assert_true(val.startswith('_') or val.endswith('_'),
+                    ('Estimator sets invalid attributes during the fit method.'
+                     ' Attributes names should either start with an _ or'
+                     ' end with a _, but we have %s' % val))
+
+
+def check_fit_changes_private_attributes_only(name, Estimator):
+    if name in ['GaussianProcess', 'GaussianProcessRegressor',
```

Comment:
This is pretty bad. Can you remove this line so that we can see the errors in continuous integration? These seems problematic and we might want to fix them.